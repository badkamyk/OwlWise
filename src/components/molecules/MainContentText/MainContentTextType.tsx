export type MainContentTextType = {
	headingText: string;
	paragraphText: string;
	headingClassName?: string;
	paragraphClassName?: string;
	secondaryText: string;
};
