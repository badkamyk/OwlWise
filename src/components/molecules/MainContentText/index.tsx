import ContentHeading from "../../atoms/ContentHeading";
import MainContentParagraph from "../../atoms/MainContentParagraph";
import { MainContentTextType } from "./MainContentTextType";

export default function MainContentText({
	headingText,
	paragraphText,
	secondaryText,
}: MainContentTextType) {
	return (
		<div
			className="font-light text-gray-500 dark:text-gray-400 sm:text-lg"
			data-testid="main-content-text"
		>
			<ContentHeading text={headingText} />
			<div className="font-normal">
				<MainContentParagraph text={paragraphText} size={"xl"} />
				<MainContentParagraph text={secondaryText} size={"xl"} />
			</div>
		</div>
	);
}
