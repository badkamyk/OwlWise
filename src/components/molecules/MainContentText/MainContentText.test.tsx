import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import MainContentText from "./index";

it("renders MainContentText with props correctly", () => {
	const { container } = render(
		<MainContentText
			headingText={"headingText"}
			paragraphText={"paragraphText"}
			secondaryText={"secondaryText"}
		/>
	);
	expect(
		container.getElementsByClassName(
			"font-light text-gray-500 dark:text-gray-400 sm:text-lg"
		)
	).toBeTruthy();
	expect(screen.getByText("headingText")).toBeTruthy();
	expect(screen.getByText("paragraphText")).toBeTruthy();
	expect(screen.getByText("secondaryText")).toBeTruthy();
	expect(container.getElementsByClassName("text-xl")).toBeTruthy();
	expect(container).toMatchSnapshot();
});
