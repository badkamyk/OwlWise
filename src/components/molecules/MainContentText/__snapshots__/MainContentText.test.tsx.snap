// Vitest Snapshot v1

exports[`MainContentText 1`] = `
<div>
  <div
    class="font-light text-gray-500 dark:text-gray-400 sm:text-lg"
  >
    <h3
      class="mb-4 tracking-tight font-extrabold text-gray-900 text-4xl"
    >
      headingText
    </h3>
    <div
      class="font-normal"
    >
      <p
        class="mb-4 text-gray-900 text-xl"
      >
        paragraphText
      </p>
      <p
        class="mb-4 text-gray-900 text-xl"
      >
        secondaryText
      </p>
    </div>
  </div>
</div>
`;
